File: Lab_2/task_3/3.swift
import Foundation

// Функция для переворачивания числа и удаления ведущих нулей
func reverseNumber(_ num: Int) -> Int {
    var reversed = 0
    var number = num
    while number > 0 {
        reversed = reversed * 10 + number % 10
        number /= 10
    }
    return reversed
}

print("Введите количество чисел: ", terminator: "")
if let nInput = readLine(), let n = Int(nInput), n > 0 {
    print("Введите \(n) чисел: ", terminator: "")

    for _ in 0..<n {
        if let numInput = readLine(), let num = Int(numInput) {
            print(reverseNumber(num), terminator: "\n")
        }
    }

    print()
} else {
    print("Некорректный ввод.")
}

File: Lab_2/task_3/3.kt
fun reverseNumber(num: Int): Int {
    var reversed = 0
    var number = num
    while (number > 0) {
        reversed = reversed * 10 + number % 10
        number /= 10
    }
    return reversed
}

fun main() {
    print("Введите количество чисел: ")
    val n = readLine()!!.toInt()

    print("Введите $n чисел: ")
    for (i in 0 until n) {
        val num = readLine()!!.toInt()
        print("${reverseNumber(num)} ")
    }

    println()
}
